File: test_regress/t/t_runflag_bad.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(vlt_all => 1);

compile(
    );

execute(
    all_run_flags => ["+verilator+bad+flag+testing"],
    fails => 1,
    expect_filename => "t/" . $Self->{name} . "_a.out",
    );

execute(
    all_run_flags => ["+verilator+rand+reset+-1"],
    fails => 1,
    expect_filename => "t/" . $Self->{name} . "_b.out",
    );

execute(
    all_run_flags => ["+verilator+rand+reset+3"],
    fails => 1,
    expect_filename => "t/" . $Self->{name} . "_c.out",
    );

execute(
    all_run_flags => ["+verilator+prof+exec+window+0"],
    fails => 1,
    expect_filename => "t/" . $Self->{name} . "_d.out",
    );

execute(
    all_run_flags => ["+verilator+prof+exec+window+1000000000000000000000000"],
    fails => 1,
    expect_filename => "t/" . $Self->{name} . "_e.out",
    );

ok(1);

1;

File: test_regress/t/t_runflag_bad.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt_all')

test.compile()

test.execute(all_run_flags=["+verilator+bad+flag+testing"],
             fails=True,
             expect_filename="t/" + test.name + "__a.out")

test.execute(all_run_flags=["+verilator+rand+reset+-1"],
             fails=True,
             expect_filename="t/" + test.name + "__b.out")

test.execute(all_run_flags=["+verilator+rand+reset+3"],
             fails=True,
             expect_filename="t/" + test.name + "__c.out")

test.execute(all_run_flags=["+verilator+prof+exec+window+0"],
             fails=True,
             expect_filename="t/" + test.name + "__d.out")

test.execute(all_run_flags=["+verilator+prof+exec+window+1000000000000000000000000"],
             fails=True,
             expect_filename="t/" + test.name + "__e.out")

test.passes()
